configure.ac

```diff
@@ -343,7 +343,13 @@ expand_path_option oldincludedir
 expand_path_option infodir
 expand_path_option mandir
 
-AS_IF([test x"${runstatedir}" = x""], [runstatedir="${sbd_runstatedir}"])
+AS_IF([test x"${runstatedir}" = x""], [runstatedir="${sbd_runstatedir}"],
+      [test x"${sbd_runstatedir}" != x""], [
+      echo "--with-runstatedir=${sbd_runstatedir} is taking precedence"
+      echo "over runstatedir=${runstatedir}."
+      echo "Consider using --runstatedir=${sbd_runstatedir} directly or let it"
+      echo "go with the default."
+      runstatedir="${sbd_runstatedir}"])
```

Comment:
I believe this will only do anything if runstatedir is empty

I think we want something like

    AS_IF([test x"${sbd_runstatedir}" != x""],
               [
                   AS_IF([test x"${runstatedir}" != x""], [echo "...taking precedence..."])
                   runstatedir="${sbd_runstatedir}"
               ])

Reply:
Well it did work in my tests.
According to documentation AS_IF is more like if ... then ... else if ... then ... else ...

Reply:
Oh right, I misread the third block as else rather than else if

looks good to me